code/code/misc/limbs.cc

```diff
@@ -1103,6 +1103,33 @@ wearSlotT pickRandomLimb(bool)
   return wearSlotT(num);
 }
 
+void TBeing::stun(spellNumT affect)
+{
+  if (task) {
+    stopTask();
+  }
+  if (spelltask) {
+    stopCast(STOP_CAST_NONE);
+  }
+
+  affectedData aff;
+  aff.type = affect;
+  aff.duration = Pulse::UPDATES_PER_MUDHOUR / 3;
+  aff.bitvector = AFF_STUNNED;
+  affectTo(&aff, -1);
+}
+
+void TBeing::stunIfLimbsChoppedOff()
+{
+  if (bothArmsHurt()) {
+    act("The pain in your crushed limbs causes you to pass out.",
```

Comment:
If you decide against my disease-based code, this message should be more abstract, since it could get called after a limb is chopped off, in which case "crushed" doesn't make sense.